### PR TITLE
CSS units for _all the pages_

### PR DESCRIPTION
**Pull request by stephencookdev**
# Problem
This is is the extension of https://github.com/onfido/onfido-sdk-ui/pull/544, but for _all the pages_!

# Solution
There are 2 minor-hacks of interest (that I've highlighted by adding a comment), but otherwise this is just the same as the previous PR.

## Checklist
- [n/a] Have the CHANGELOG, README, MIGRATION, RELEASE_GUIDELINES docs been updated?
- [n/a] Have new automated tests been implemented?
- [n/a] Have new manual tests been written down?
- [n/a] Have tests passed locally?
- [n/a] Have any new strings been translated?

### CODE REVIEW

File: src/components/CameraPermissions/Primer/style.css [hunk 1]
```diff
@@ -1,47 +1,52 @@
 @import (less) "../../Theme/constants.css";
 
-.text {
-  margin-top: 64px;
-  margin-bottom: 80px;
-}
+@font-size: 14;
+@font-size-units: @font-size*@unit;
+@sub-unit: 1em/@font-size;
```
Comment (stephencookdev) [inline src/components/CameraPermissions/Primer/style.css]: This is the first "hack" that's a bit weird. Just creating some `sub-unit`s on the fly, since we change the font-size and it confuses things

File: src/components/Theme/style.css [hunk 2]
```diff
@@ -108,6 +107,20 @@
   border-radius: 32*@unit;
   padding: 0;
   outline: none;
+
+  /*
+   * Beware! Hack!
```
Comment (stephencookdev) [inline src/components/Theme/style.css]: This is the second hack... That really is a hack :/

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-552-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-552-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://552-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/Collapsible/style.css [hunk 3]
```diff
@@ -10,13 +12,13 @@
   background-repeat: no-repeat;
   content: '';
   display: inline-block;
-  height: 6px;
-  margin-left: 5px;
+  height: 7*@unit;
```
Comment (stefaniacardenas) [inline src/components/Collapsible/style.css]: Why did you add 1 to all the values in the file?

Comment (stephencookdev) [reply src/components/Collapsible/style.css]: The `Collapsible` component is only used in 1 place, and it's in a place with `font-size: 14px` :sob:

I set new values that make the component look the same in that context

File: src/components/CameraPermissions/Primer/style.css (hunk 1 above)
Comment (stefaniacardenas) [reply src/components/CameraPermissions/Primer/style.css]: Could you use a less mixin here to always make sure that unit returns the right value based on the actual font-size. And use the default value to calculate @font-size-units?

Comment (stephencookdev) [reply src/components/CameraPermissions/Primer/style.css]: I don't we can in LESS 😢

File: src/components/Theme/style.css (hunk 2 above)
Comment (stephencookdev) [reply src/components/Theme/style.css]: I've refactored this out now, so that we have a `Button` component, which has a text span, so we no longer have this hack!

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-552-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-552-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-552-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-552-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/Confirm/index.js [hunk 4]
```diff
@@ -9,24 +9,27 @@ import { preventDefaultOnClick } from '~utils/index'
...
+    className={style.retake}
+    variants={["outline"]}
+  >
     {translate('confirm.redo')}
-  </button>
+  </Button>
 )
 
 const ConfirmAction = localised(({confirmAction, translate, error}) =>
-  <button href='#' className={`${theme.btn} ${theme["btn-primary"]}`}
```
Comment (stefaniacardenas) [inline src/components/Confirm/index.js]: Removing this class `theme["btn-primary"]` will make the tests fail because cucumber will be looking for this class in order to proceed to the next screen

Comment (stephencookdev) [reply src/components/Confirm/index.js]: Ahh, I missed that. I've fixed up the E2E test references so that they refer to the new expected class name :)

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-552-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://552-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-552-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-552-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-552-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://552-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-552-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-552-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-552-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-552-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://552-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/PhoneNumberInput/style.css [hunk 5]
```diff
@@ -1,34 +1,79 @@
...
-.phoneNumberContainer :global .rrui__select__options:not(.rrui__select__options--menu) {
-  width: 438px;
+  .rrui__select__options:not(.rrui__select__options--menu) {
+    width: 438*@unit;
+  }
+
+  /* We need to override the `rrui` styles from REM to EM, for our relative fonts */
+  .rrui__input:not(.rrui__input--multiline),
+  .rrui__input-label {
+    height: 1.8em;
```
Comment (rfreitas) [inline src/components/PhoneNumberInput/style.css]: hm, shouldn't these have the @unit indirection as well?

File: src/components/PrivacyStatement/style.css [hunk 6]
```diff
@@ -15,72 +15,69 @@
...
+  margin-bottom: 24*@unit;
   @media (--small-viewport) {
     margin-bottom: 0;
   }
 }
 
 .list {
-  font-size: 16px;
-  line-height: 24px;
+  line-height: 1.5;
```
Comment (rfreitas) [inline src/components/PrivacyStatement/style.css]: this is a quite common, I wonder if we should make it as past of a text base mixin, what do you think?

File: src/components/crossDevice/CrossDeviceLink/style.css [hunk 7]
```diff
@@ -101,29 +101,29 @@
...
 .linkText {
   resize: none;
-  height: 24px;
+  height: 24*@unit;
   border: none;
   background-color: #f3f3f4;
   overflow: hidden;
-  font-size: 16px;
-  line-height: 24px;
+  font-size: inherit;
```
Comment (rfreitas) [inline src/components/crossDevice/CrossDeviceLink/style.css]: curious, why is the inherit necessary?

File: src/components/Video/style.css [hunk 8]
```diff
@@ -102,33 +106,36 @@
...
+    margin-right: 5*@unit;
     background: #DC2A2A;
-    border-radius: 4px;
+    border-radius: 4*@unit;
     line-height: 120%;
   }
 }
+.recordingIndicatorText {
+  font-size: 11*@unit;
+  line-height: 1.45;
```
Comment (rfreitas) [inline src/components/Video/style.css]: curious: this seems to be tantalisingly close to `line-height: 1.5`, what is the reason?

File: src/components/PhoneNumberInput/style.css (hunk 5 above)
Comment (stephencookdev) [reply src/components/PhoneNumberInput/style.css]: I'm trying to keep them as close to the library's styles that we're overriding as possible

The library was doing stuff like `calc(10px + 1em)` for reasons I don't know. We could use `@unit` I guess, I was just trying to make it remain "other"

File: src/components/PrivacyStatement/style.css (hunk 6 above)
Comment (stephencookdev) [reply src/components/PrivacyStatement/style.css]: How do you mean? `line-height: 1.5` you mean?
I think we have quite a few different line-heights that we use

File: src/components/ProofOfAddress/Guidance/style.css [hunk 9]
```diff
@@ -2,29 +2,33 @@
...
 
   @media (--small-viewport) {
-    margin: 0 24px 1rem;
+    margin: 0 24*@unit 16*@unit;
   }
 }
 
 .subTitle {
-  font-size: 14px;
+  @font-size: 14;
```
Comment (rfreitas) [inline src/components/ProofOfAddress/Guidance/style.css]: this logic seems to be used in multiple places. seem like something which is worth abstracting perhaps?

File: src/components/crossDevice/CrossDeviceLink/style.css (hunk 7 above)
Comment (stephencookdev) [reply src/components/crossDevice/CrossDeviceLink/style.css]: Some input elements like `button` are annoying in that they don't default in inheriting their parents fonts

Here we're just making 100% sure that we _are_ inheriting our parent's fonts, in case we end up in such an awkward element

File: src/components/Video/style.css (hunk 8 above)
Comment (stephencookdev) [reply src/components/Video/style.css]: Just keeping the design pixel-perfect to the SDK as it stands now, where possible

File: src/components/ProofOfAddress/Guidance/style.css (hunk 9 above)
Comment (stephencookdev) [reply src/components/ProofOfAddress/Guidance/style.css]: I agree. Sadly, I don't think LESS gives us any such capability to abstract :(

File: src/components/PrivacyStatement/style.css (hunk 6 above)
Comment (rfreitas) [reply src/components/PrivacyStatement/style.css]: yep, `line-height: 1.5` is used in 5 places, hence why I wonder if it could be part of a standard text base. but no strong feelings about it tbh

File: src/components/Video/style.css (hunk 8 above)
Comment (rfreitas) [reply src/components/Video/style.css]: I guess, you figured out what was the defacto lineheight when it was all still `px`?

File: src/components/ProofOfAddress/Guidance/style.css (hunk 9 above)
Comment (rfreitas) [reply src/components/ProofOfAddress/Guidance/style.css]: I see 😢

Comment (rfreitas) [reply src/components/ProofOfAddress/Guidance/style.css]: I guess we can still abstract in a crude way, since it's always `@font-size: 14`.

```
@smaller-font-size-unitless: 14

//to export
@smaller-font-size: @@smaller-font-size-unitless*@unit;
@smaller-font-unit: 1em/@smaller-font-size
```

File: src/components/PhoneNumberInput/style.css (hunk 5 above)
Comment (stefaniacardenas) [reply src/components/PhoneNumberInput/style.css]: I think they do that to handle the round icon used for international and the square icon used for specific country flags

File: src/components/ProofOfAddress/Guidance/style.css (hunk 9 above)
Comment (stephencookdev) [reply src/components/ProofOfAddress/Guidance/style.css]: Good idea, I've made this change now 🙂 
We now have:

```
@font-size-base
@unit

@font-size-small
@unit-small

@font-size-x-small
@unit-x-small
```

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-552-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-552-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://552-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-552-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://552-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-552-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/PrivacyStatement/style.css [hunk 10]
```diff
@@ -15,72 +15,69 @@
...
-    padding: 0px 16px;
+    width: 112*@unit;
+    padding: 0 16*@unit;
   }
 }
 
 .primary {
   @media (--small-viewport) {
-    width: 152px;
+    width: 152*@unit;
```
Comment (stefaniacardenas) [inline src/components/PrivacyStatement/style.css]: I think we need to do something like here for the confirm screen primary button, because at the moment, on small screens it tries to fill all the space between the two buttons.

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-552-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://552-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-552-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/Theme/style.css [hunk 11]
```diff
@@ -173,29 +104,23 @@
   }
 }
 
-.actions .btn-primary {
```
Comment (stefaniacardenas) [inline src/components/Theme/style.css]: This change is affecting the layout of the primary button in the confirm screen

Comment (rfreitas) [reply src/components/Theme/style.css]: got it!

Comment (rfreitas) [reply src/components/Theme/style.css]: fixed in commit https://github.com/onfido/onfido-sdk-ui/pull/552/commits/4eb4616d59ab86d1184c8603dc3a013d346eb0fe

and nice catch btw!

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-552-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://552-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-552-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/Theme/style.css [hunk 12]
```diff
@@ -94,16 +94,6 @@
   height: 64*@unit;
 }
 
-.actions {
```
Comment (stefaniacardenas) [inline src/components/Theme/style.css]: Isn't this class shared with other components?

File: src/components/Confirm/index.js [hunk 13]
```diff
@@ -9,30 +8,35 @@ import { preventDefaultOnClick } from '~utils/index'
...
+  <Button
+    className={style["btn-primary"]}
+    variants={["primary"]}
```
Comment (stefaniacardenas) [inline src/components/Confirm/index.js]: I can see that the style of this class is overriding the style in `style.button-primary`.

File: src/components/Theme/style.css (hunk 12 above)
Comment (rfreitas) [reply src/components/Theme/style.css]: no

File: src/components/Confirm/index.js (hunk 13 above)
Comment (rfreitas) [reply src/components/Confirm/index.js]: yes, that's the idea. what is your concern?

Comment (stefaniacardenas) [reply src/components/Confirm/index.js]: The button on small screens should have [this width](https://github.com/onfido/onfido-sdk-ui/blob/4eb4616d59ab86d1184c8603dc3a013d346eb0fe/src/components/Confirm/style.css#L43) instead of [this one](https://github.com/onfido/onfido-sdk-ui/blob/4eb4616d59ab86d1184c8603dc3a013d346eb0fe/src/components/Button/style.css#L40). 
screen shot 2019-02-18 at 12 08 57

Comment (rfreitas) [reply src/components/Confirm/index.js]: got it now! I actually thought you meant the opposite. On my side I didn't see the bug, but I tested it on webpack dev server, which probably reorders the styles. Nonetheless, they both have the same specificity, so it needs to be fixed! Good catch.

fix here: https://github.com/onfido/onfido-sdk-ui/pull/552/commits/26871d33f0135f156c8841097c7f81ed3bb81cc5

Comment (rfreitas) [reply src/components/Confirm/index.js]: btw, do you know why `button-primary` has 
```
@media (--small-viewport) {
    width: 100%;
  }
``` ? It feels like a layout logic, probably better placed outside of the button.

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-552-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/Confirm/index.js (hunk 13 above)
Comment (stefaniacardenas) [reply src/components/Confirm/index.js]: On small screens when there is only one action, the button takes 100% of the width (see Welcome screen and video intro screen)

**Comment by rfreitas** [comment]
Travis automatic deployment: https://552-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-552-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/Confirm/index.js (hunk 13 above)
Comment (rfreitas) [reply src/components/Confirm/index.js]: I see. I wonder whether we should move that logic outside of the button at some point then. Feels like a job of the parent to decide if the button takes the entire width.

Comment (stefaniacardenas) [reply src/components/Confirm/index.js]: Yes, I agree. Not ideal to have this logic in the theme.

Comment (rfreitas) [reply src/components/Confirm/index.js]: it's not in the theme. it's in the Button/style.css

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-552-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://552-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-552-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-552-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://552-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-552-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-552-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-552-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://552-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-552-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-552-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://552-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-552-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-552-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://552-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-552-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-552-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://552-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-552-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://552-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-552-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-552-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-552-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://552-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-552-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-552-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-552-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-552-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://552-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-552-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-552-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://552-pr-onfido-sdk-ui-onfido.surge.sh
